webpack.config.js

```diff
@@ -9,9 +10,15 @@ module.exports = {
   output: {
     path: 'dist',
     filename: '[name].js',
+    library: 'synaptic',
+    libraryTarget: 'umd'
   },
   plugins: [
     new webpack.NoErrorsPlugin(),
-    new webpack.BannerPlugin(license())
+    new webpack.BannerPlugin(license()),
+    new webpack.optimize.UglifyJsPlugin({
```

Comment:
The reason we were not minifying was because that was breaking the `trainAsync` methods. You can test it if you add an `index.html` file in the repo root directory with the following:

```html
<script src="./dist/synaptic.min.js"></script>
<script>
  var net = new synaptic.Architect.Perceptron(2, 4, 1);
  var trainer = new synaptic.Trainer(net);
  trainer.trainAsync([
    {
      input: [0, 0],
      output: [0]
    },
    {
      input: [0, 1],
      output: [1]
    },
    {
      input: [1, 0],
      output: [1]
    },
    {
      input: [1, 1],
      output: [0]
    }
  ], { log: 1 });

</script>
```

You will see an error in the console, and if you change the `src` to `./dist/synaptic.js` it will work again.

This is something that won't happen anymore in the next version, but I don't think it will get fixed on this one, so for now we should remove this plugin from the build.